core/finality-grandpa/src/tests.rs

```diff
@@ -190,6 +205,27 @@ impl Network for MessageRouting {
 			gossip.collect_garbage(|t| t == &topic)
 		});
 	}
+
+	fn commit_messages(&self, set_id: u64) -> Self::In {
+		let inner = self.inner.lock();
+		let peer = inner.peer(self.peer_id);
+		let mut gossip = peer.consensus_gossip().write();
+		let messages = peer.with_spec(move |_, _| {
+			gossip.messages_for(make_commit_topic(set_id))
+		});
+
+		let messages = messages.map_err(
+			move |_| panic!("Commit messages for set {} dropped too early", set_id)
```

Comment:
Is this `Stream` running on a tokio runtime? Are you aware that tokio catches `panics!` and continues?

Reply:
That's not true for `current_thread::block_on`, which these tests all use.

Reply:
There's nothing in the type signature of `block_on` to accommodate panics